src/GovUk.Education.ExploreEducationStatistics.Data.Services/ObservationService.cs

```diff
@@ -160,6 +81,117 @@ public IEnumerable<Observation> FindObservations(SubjectMetaQueryContext query)
+                var wardListParam =
+                    CreateIdListType("wardList", locationsQuery?.Ward);
+                var filterItemListParam = CreateIdListType("filterItemList", query.Filters);
+
+                // EES-745 It's ok to use Observation as the return type here, as long as only the Id field is selected
+
+                var inner = context
+                    .Set<Observation>()
+                    .FromSqlRaw("EXEC dbo.FilteredObservations " +
+                                "@subjectId," +
+                                "@geographicLevel," +
+                                "@timePeriodList," +
+                                "@countriesList," +
+                                "@englishDevolvedAreaList," +
+                                "@institutionList," +
+                                "@localAuthorityList," +
+                                "@localAuthorityOldCodeList," +
+                                "@localAuthorityDistrictList," +
+                                "@localEnterprisePartnershipList," +
+                                "@mayoralCombinedAuthorityList," +
+                                "@multiAcademyTrustList," +
+                                "@opportunityAreaList," +
+                                "@parliamentaryConstituencyList," +
+                                "@providersList," +
+                                "@regionsList," +
+                                "@rscRegionsList," +
+                                "@schoolsList," +
+                                "@sponsorList," +
+                                "@wardList," +
+                                "@planningAreaList," +
+                                "@filterItemList",
+                        subjectIdParam,
+                        geographicLevelParam,
+                        timePeriodListParam,
+                        countriesListParam,
+                        englishDevolvedAreaListParam,
+                        institutionListParam,
+                        localAuthorityListParam,
+                        localAuthorityOldCodeListParam,
+                        localAuthorityDistrictListParam,
+                        localEnterprisePartnershipListParam,
+                        mayoralCombinedAuthorityListParam,
+                        multiAcademyTrustListParam,
+                        opportunityAreaListParam,
+                        parliamentaryConstituencyListParam,
+                        providersListParam,
+                        regionsListParam,
+                        rscRegionListParam,
+                        schoolsListParam,
+                        sponsorListParam,
+                        wardListParam,
+                        planningAreaListParam,
+                        filterItemListParam);
+
+                return await inner
+                    .Select(obs => obs.Id)
+                    .ToArrayAsync(cancellationToken);
```

Comment:
Interested to know if this will stop execution of the stored procedure on the database server or if we need to make any further changes there to make it respect a cancellation?

Reply:
It does!  Very hard to unit test that though without guaranteeing a real DB connection.

Reply:
Great to hear 🙂